app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -20,6 +23,10 @@ def index
         enabled_mfa_methods_count:,
         gov_or_mil_email: fed_or_mil_email?,
       )
+      @tmx_view_needed =
+        FeatureManagement.account_creation_device_profiling_collecting_enabled? &&
+        mfa_context.enabled_mfa_methods_count == 0
```

Comment:
`enabled_mfa_methods_count` spawns a lot of different database queries. Is there any harm to just let the profiling script run even if it's their second time on the screen?

Reply:
I guess theres no harm, I wanted to reduce us making unneeded calls to tmx with this. but im open to just let it happen.

Reply:
I think deleting the session value would just regenerate the session id again, As for the session value, totally possible but then we would need to pass the session in so that we can change or update that user session. If we don't believe theres much of a reason to prevent us reloading the javascript polling again, we can make it simple for now.